pkg/rest/config_upload.go

```diff
@@ -29,28 +29,38 @@ import (
 	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util"
 )
 
-func upsertDynatraceObject(client *http.Client, environmentUrl string, objectName string, theApi api.Api, payload []byte, apiToken string) (api.DynatraceEntity, error) {
-
+func upsertDynatraceObject(
+	client *http.Client,
+	environmentUrl string,
+	objectName string,
+	theApi api.Api,
+	payload []byte,
+	apiToken string,
+) (api.DynatraceEntity, error) {
 	fullUrl := theApi.GetUrlFromEnvironmentUrl(environmentUrl)
-
-	existingObjectId, err := getObjectIdIfAlreadyExists(client, theApi, fullUrl, objectName, apiToken)
-	if err != nil {
-		return api.DynatraceEntity{}, err
-	}
-
 	body := payload
 	configType := theApi.GetId()
+	isLegacyApi := theApi.IsLegacyApi()
+	objectId := ""
+
+	// The calculated-metrics-log  and legacy APIs don't have a POST endpoint.
+	// To create a new log metric we need to use PUT which requires a metric key
+	// for which we can just take the objectName
+	if isLegacyApi || configType == "calculated-metrics-log" {
+		objectId = objectName
```

Comment:
That is actually not true is it? The legacy APIs do not require an identifier (unlike metrics). 

Your change to the update method actually drops the id again. 

I'd prefer if you move the building of the update API path out of the upload completely, then we have one less "if legacy do something else". 

---

Additionally you've changed the update logic for calculated-metrics-log!  **if** a config object already exists with the configured name, and an ID that is not equal to the name, you will create a new one rather than update. 
I'm not 100% sure if that is an issue right now as I don't know what will be set as ID when such configs are created in the UI - it might be fine with that also uses the name but it will be an issue if generates a UUID. Also I would not rely on any assumptions on how Dynatrace behaves internally in this case.